sync/class.jetpack-sync-module-themes.php

```diff
@@ -23,6 +25,44 @@ public function init_listeners( $callable ) {
 		add_action( 'jetpack_widget_reordered', $callable );
 	}
 
+	public function detect_theme_edit( $redirect_url ) {
+		$url = wp_parse_url( admin_url( $redirect_url ) );
```

Comment:
$redirect_url is just the file name when/where we need to examine it, see https://github.com/WordPress/WordPress/blob/master/wp-admin/theme-editor.php#L124